### PR TITLE
Do not load banner till dismiss count is available.

### PR DESCRIPTION
**Pull request by ankitrox**
## Summary

Addresses issue:

- #9231

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js [hunk 1]
```diff
@@ -504,10 +534,27 @@ describe( 'AudienceSegmentationSetupCTAWidget', () => {
...
+						data: {
+							slug: AUDIENCE_SEGMENTATION_SETUP_CTA_NOTIFICATION,
+							expiration: 0,
+						},
+					},
+					method: 'POST',
+				} );
+			} );
+
 			await act( () => waitForTimeouts( 30 ) );
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js:558]: We can verify the widget is no longer rendered, and remove the `waitForTimeouts()` at the end:

```suggestion

				expect( container ).toBeEmptyDOMElement();
			} );
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js [hunk 2]
```diff
@@ -270,6 +276,30 @@ describe( 'AudienceSegmentationSetupCTAWidget', () => {
...
+				{
+					registry,
+				}
+			);
+
+			expect(
+				queryByText(
+					/Learn how different types of visitors interact with your site/i
+				)
+			).not.toBeInTheDocument();
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js:301]: I'd suggest checking `container` is empty, it's a bit more of a complete test:

```suggestion
			const { container } = render(
				<AudienceSegmentationSetupCTAWidget
					Widget={ Widget }
					WidgetNull={ WidgetNull }
				/>,
				{
					registry,
				}
			);

			expect( container ).toBeEmptyDOMElement();
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js [hunk 3]
```diff
@@ -105,7 +111,13 @@ function AudienceSegmentationSetupCTAWidget( { Widget, WidgetNull } ) {
 
 	const { apiErrors, failedAudiences, isSaving, onEnableGroups } =
 		useEnableAudienceGroup( {
-			onSuccess: () => {
+			onSuccess: async () => {
+				await dismissPrompt(
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js:115]: Hmm, maybe we should _not_ make this async so the dismissal requests can potentially fire in parallel:

```suggestion
			onSuccess: () => {
				dismissPrompt(
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js [hunk 4]
```diff
@@ -88,6 +88,12 @@ function AudienceSegmentationSetupCTAWidget( { Widget, WidgetNull } ) {
 		)
 	);
 
+	const dismissCountLoaded = useSelect( ( select ) =>
+		select( CORE_USER ).hasFinishedResolution( 'getPromptDismissCount', [
+			AUDIENCE_SEGMENTATION_SETUP_CTA_NOTIFICATION,
+		] )
+	);
+
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js:96]: `getPromptDismissCount` doesn't have a resolver, so it doesn't make sense to try to wait for it to resolve - `dismissCountLoaded` will only ever be `false`. We can instead check the `getDismissedPrompts` resolver.

```suggestion
	const dismissedPromptsLoaded = useSelect( ( select ) =>
		select( CORE_USER ).hasFinishedResolution( 'getDismissedPrompts', [] )
	);
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js [hunk 5]
```diff
@@ -185,7 +197,8 @@ function AudienceSegmentationSetupCTAWidget( { Widget, WidgetNull } ) {
 		configuredAudiences === undefined ||
 		configuredAudiences?.length ||
 		! analyticsIsDataAvailableOnLoad ||
-		isDismissed
+		isDismissed ||
+		! dismissCountLoaded
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js:201]: As per the above name change:

```suggestion
		! dismissedPromptsLoaded
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js [hunk 6]
```diff
@@ -126,6 +126,12 @@ describe( 'AudienceSegmentationSetupCTAWidget', () => {
 			.dispatch( MODULES_ANALYTICS_4 )
 			.finishResolution( 'getReport', [ options ] );
 
+		registry
+			.dispatch( CORE_USER )
+			.finishResolution( 'getPromptDismissCount', [
+				AUDIENCE_SEGMENTATION_SETUP_CTA_NOTIFICATION,
+			] );
+
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js:134]: As above, `getPromptDismissCount` doesn't have a resolver so it doesn't make sense to try to interact with it.

You should instead populate and resolve `getDismissedPrompts()`.

```suggestion
		const currentTimeInSeconds = Math.floor( Date.now() / 1000 );

		registry.dispatch( CORE_USER ).receiveGetDismissedPrompts( {
			[ AUDIENCE_SEGMENTATION_SETUP_CTA_NOTIFICATION ]: {
				expires: currentTimeInSeconds - 10,
				count: 0,
			},
		} );

		registry
			.dispatch( CORE_USER )
			.finishResolution( 'getDismissedPrompts', [] );
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js [hunk 7]
```diff
@@ -270,6 +276,30 @@ describe( 'AudienceSegmentationSetupCTAWidget', () => {
 				)
 			).not.toBeInTheDocument();
 		} );
+
+		it( 'should not render the widget when the prompt dismiss count is not resolved', () => {
+			registry
+				.dispatch( CORE_USER )
+				.startResolution( 'getPromptDismissCount', [
+					AUDIENCE_SEGMENTATION_SETUP_CTA_NOTIFICATION,
+				] );
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js:285]: As above:

```suggestion
				.startResolution( 'getDismissedPrompts', [] );
```

**Review by techanvil — CHANGES_REQUESTED** [review]
Thanks @ankitrox, a few changes are needed here - please see my comments.

**Comment by ankitrox** [comment]
Thanks so much @techanvil for looking into this and adding the feedback.

I've made the changes and assigning back to you for re-review.

**Review by techanvil — APPROVED** [review]
That's great, thanks @ankitrox. LGTM!

**Pull request merged**